File: NHSOnline.Backend.Worker.Mocking/Emis/Models/ExceptionResponse.cs
using System.Collections.Generic;

namespace NHSOnline.Backend.Worker.Mocking.Emis.Models
{
    public class ExceptionResponse
    {
        public long InternalResponseCode { get; }
        public string InternalResponseMessage => "Exception occurred during API processing.";
        public IEnumerable<Exception> Exceptions { get; }


        public ExceptionResponse(long internalResponseCode, string exceptionMessage)
        {
            InternalResponseCode = internalResponseCode;
            Exceptions = new[] { new Exception(exceptionMessage) };
        }

        public class Exception
        {
            public string CallStack { get; }
            public string ExceptionId { get; }
            public string InnerExceptionId { get; }
            public bool IsRootException { get; }
            public string Message { get; }

            public Exception(string message)
            {
                CallStack = "Not available";
                ExceptionId = "00000000-0000-0000-0000-000000000000";
                InnerExceptionId = "00000000-0000-0000-0000-000000000000";
                IsRootException = true;
                Message = message;
            }
        }
    }


}

File: bddtests/src/main/kotlin/mocking/models/ExceptionResponse.kt
package mocking.microtest.models

class ExceptionResponse(val internalResponseCode: Long,
                        exceptionMessage: String) {
    val internalResponseMessage = "Exception occurred during API processing."
    val exceptions = arrayOf(Exception(exceptionMessage))


    class Exception(val message: String) {
        val callStack = "Not available"
        val exceptionId = "00000000-0000-0000-0000-000000000000"
        val innerExceptionId = "00000000-0000-0000-0000-000000000000"
        val isRootException = true
    }
}